### DISCUSSION TITLE
Is there an API to search identity by email address?

### DISCUSSION THREAD

**Discussion by frederikhors** [ory/kratos#2379] [Q&A] [answered]
I'm using a default schema with `email` as `identifier` and the below code to "create" new identity:

```go
identityBody := kratos.NewAdminCreateIdentityBody("default", map[string]interface{}{
  "email":     input.Email(),
  "customTrait": customTrait,
})

identity, httpResp, err := r.kratos.V0alpha2Api.AdminCreateIdentity(ctx).AdminCreateIdentityBody(*identityBody).Execute()
```

**IT WORKS AMAZIGLY!**

### Question

Sometimes there are duplicate email addresses and of course kratos returns an error with code 409. Great.

What I would like, however, is to find the ID of that duplicate email and it would be amazing to have that ID in 409 response (for ADMIN API only).

There is currently no way to do this, right?

One way I think is to call the "Admin List identitites" method and `range` on the result until I find the email in the traits?

Obviously it is a very slow method compared to a dedicated API.

**I'm doing it wrong?** Are new APIs planned?

**Comment by aeneasr** [comment]
Hello @frederikhors, such an API is indeed in the works: https://github.com/ory/kratos/pull/1671

**Comment by ghost** [comment]
http://127.0.0.1:4434/admin/identities?credentials_identifier=stone+50@gmail.com

Kratos does not return user even with above email user exist but + sign in email creating issue to find user I have checked anybody know how to find user with email contain + sign.

**Reply by jonas-jonas** [reply]
You might need to escape the `+` sign with `%2B`. Alternatively, use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI to encode the entire URL.

**Reply by ghost** [reply]
Thanks I have tried ory copilot and it return same response

**Comment by CreMindES** [comment]
Just to spare other people from going through 3 PRs, please see https://github.com/ory/kratos/issues/2783.

**Reply by vinckr** [reply, marked as answer]
See the changelog entry: https://changelog.ory.sh/announcements/an-api-to-fetch-identities-by-email-or-username
